backend/errors/error_source.go

```diff
@@ -0,0 +1,37 @@
+package errors
+
+import (
+	"net/http"
+)
+
+func GetErrorSource(statusCode int) ErrorSource {
+	switch statusCode {
+	case http.StatusBadRequest,
+		http.StatusMethodNotAllowed,
+		http.StatusNotAcceptable,
+		http.StatusPreconditionFailed,
+		http.StatusRequestEntityTooLarge,
+		http.StatusRequestHeaderFieldsTooLarge,
+		http.StatusRequestURITooLong,
+		http.StatusExpectationFailed,
+		http.StatusUpgradeRequired,
+		http.StatusRequestedRangeNotSatisfiable,
+		http.StatusNotImplemented:
+		return ErrorSourcePlugin
+	}
+
+	if statusCode < 400 {
+		return ErrorSourceNone
+	}
+
+	return ErrorSourceDownstream
+}
+
+// ErrorSource type defines the source of the error
+type ErrorSource string
+
+const (
+	ErrorSourcePlugin     ErrorSource = "plugin"
+	ErrorSourceDownstream ErrorSource = "downstream"
+	ErrorSourceNone       ErrorSource = "none"
```

Comment:
For simplicity can we merge None with Plugin or what would be the use case of this None status?

Reply:
I think you earlier suggested that we didn't want to have an empty error source.  I saw [this in core grafana](https://github.com/grafana/grafana/blob/main/pkg/tsdb/prometheus/instrumentation/instrumentation.go#L27).

Reply:
removed, but still curious we we have "none" in core grafana

Reply:
We have "none" for the Prometheus datasource in core Grafana, not for the request HTTP metrics where it could only be _server_ or _downstream_ - I don't know what their use case of it is. There might be differences so would be interesting to know what "none" means in Prometheus